hoomd/md/update.py

```diff
@@ -149,133 +150,134 @@ class mueller_plathe_flow:
     "min" slap might be swapped.
 
     Args:
-        group (``hoomd.group``): Group for which the update will be set
-        flow_target (`hoomd.variant`): Integrated target flow. The unit is the
-            in the natural units of the simulation: [flow_target] = [timesteps]
-            x :math:`\mathcal{M}` x :math:`\frac{\mathcal{D}}{\tau}`.  The unit
-            of [timesteps] is your discretization dt x :math:`\mathcal{\tau}`.
-        slab_direction (``X``, ``Y``, or ``Z``): Direction perpendicular to the
-            slabs..
-        flow_direction (``X``, ``Y``, or ``Z``): Direction of the flow..
-        n_slabs (int): Number of slabs. You want as many as possible for small
-            disturbed volume, where the unphysical swapping is done. But each
-            slab has to contain a sufficient number of particle.
+        filter (`hoomd.filter.ParticleFilter`): Subset of particles on which to
+            apply this updater.
+
+        flow_target (`hoomd.variant.Variant`): Integrated target flow.
+            :math:`[\\delta t \\cdot \\mathrm{mass} \\cdot \\mathrm{length}
```

Comment:
I updated the style here so that it renders correctly. This is a direct translation of the old documentation. It doesn't make sense to me because `\delta t \cdot \mathrm{time}^{-1}` cancels out and there is no time unit here.

Reply:
I don't get this unit either. I am not perfectly familiar with Mueller-Plathe flow but ussually the units of shear flow are `N \cdot \mathrm{length}^{-1} = \mathrm{mass} \cdot \mathrm{time}^{-2}`.

Reply:
I think the units should either be `mass * length^(-1)` or `mass * time^(-1) * length^(-1)`.

I believe the `flow_target` argument is defining a target value for the _time-integrated momentum flux_, because in the paper where this method was implemented https://doi.org/10.1016/j.commatsci.2019.109107 they say:

> Therefore we have chosen to control the time-integrated momentum flux

Since momentum flux is in units of `mass * length * time^(-1) * length^(-2)` I think the time-integrated momentum flux should be in units of `delta_t * mass * time^(-1) * length^(-1)` or just `mass * length^(-1)`.

However, it looks like the code is expecting something in units of `mass * length * time^(-1) * length^(-2)` or `mass * time^(-1) * length^(-1)`. This is the line of code that is using flow_target:

`while (fabs((*m_flow_target)(timestep) - this->getSummedExchangedMomentum()/area) > this->getFlowEpsilon())`

where `getSummedExchangedMomentum()` returns `m_exchanged_momentum` which is found in this line:

`m_exchanged_momentum += sign * (m_last_max_vel.x - m_last_min_vel.x);`

where `m_last_max_vel` is actually storing the momentum, and `sign` is just +/- 1 based on how the min and max slabs are defined.

So it looks like the code is summing momenta and dividing the sum by an area, and then comparing this to `flow_target`. I don't think it is multiplying by `dt` during this sum.